cmd/tealdbg/localLedger.go

```diff
@@ -285,11 +285,36 @@ func (l *localLedger) CheckDup(config.ConsensusParams, basics.Round, basics.Roun
 	return nil
 }
 
-func (l *localLedger) LookupWithoutRewards(rnd basics.Round, addr basics.Address) (basics.AccountData, basics.Round, error) {
+func (l *localLedger) LookupResource(rnd basics.Round, addr basics.Address, aidx basics.CreatableIndex, ctype basics.CreatableType) (ledgercore.AccountResource, error) {
```

Comment:
I notice that nearly every "top-level" caller of LookupResource, knows the type it wants. I think it would make all these callers clearer if the Ledger interface offered `LookupAsset` and `LookupApp` instead, which would not need the extra argument, and would be typed more accurately (they would take the specific index type instead of basics.CreateableIndex.) They would also return a more specific type than `ledgercore.AccountResource`.

Are there callers that need to call LookupResource and they don't know what they want?

Reply:
I do like the idea, but I think that we should hold off with that type of changes due to their cascading consequences. Changing LookupResource to LookupAsset+LookupApp suggest that the return value would also need to be modified to be class-specific. This would have trailing effects that I'm not sure if we really want to explore at this time.
Let's create a PR for that which we could implement once this feature is shipped ?

Reply:
created a follow up issue to work on this issue.